monkey/infection_monkey/dropper.py

```diff
@@ -45,7 +46,7 @@ def __init__(self, args):
         arg_parser.add_argument("-p", "--parent")
         arg_parser.add_argument("-t", "--tunnel")
         arg_parser.add_argument("-s", "--server")
-        arg_parser.add_argument("-d", "--depth", type=int)
+        arg_parser.add_argument("-d", "--depth", type=positive_int, default=0)
```

Comment:
This is a bit confusing, because `positive_int` does not allow the value to be `0`, but we set `default=0`.

Someone could conceivably set the depth to zero if they just want to run a payload on a single machine and want extra assurance that the agent doesn't spread. We should change `positive_int` to a different type that allows 0.

What about `z_nonneg`? 

> Z<sup>nonneg</sup> is the set of all positive integers including 0
"https://calculus.nipissingu.ca/tutorials/numbers.html

Reply:
That article is self-contradictory:

![image](https://user-images.githubusercontent.com/19957806/175031961-c61dc320-4130-422b-875c-172943c68f46.png)

![image](https://user-images.githubusercontent.com/19957806/175031782-785454e4-6ef4-4dd4-8f08-c889579886c7.png)